### ISSUE TITLE
Nessie OAuth2 authentication

### ISSUE THREAD

**Issue by tmnd1991** [projectnessie/nessie#4563]
**Description**: Reading Nessie docs I understand that the only option (when running nessie on k8s) for authentications is using OIDC. Can you confirm that AWS authentication mode is restricted to Nessie deployed as lambdas?
I also understand that Nessie piggybacks on quarkus. I see that quarkus has support for [oauth2](https://quarkus.io/guides/security-oauth2), but I guess I would need to rebuild Nessie itself. 

Would it be possible to have a Nessie image with `quarkus-elytron-security-oauth2` extension?

**Expected Use Cases**:
Perform authentication through OAuth2 on a Nessie server.

**Comment by snazy** [comment]
Yes, building Nessie with that extension is currently the only way to get it in.
Want to take a stab on a PR adding it?

**Comment by tmnd1991** [comment]
Yup. How would you go for it? Create a new gradle 
 module which includes nessie-server and adds that dependency?

**Comment by snazy** [comment]
Nah - not another nessie server variant. Would prefer to add it to the existing one and have it configurable.

**Comment by snazy** [comment]
@dimas-b is this something we need to consider for API v2?

**Comment by dimas-b** [comment]
I _guess_ this is not related to the REST API rework. I believe all auth stuff is rather independent of actual REST parameters. This is more of a client thing, I guess.
